### PR TITLE
feat(init-capital): deploy and set tokens for api3 proxy oracle reader

### PR DESCRIPTION
**Pull request by mewpawee**

### CODE REVIEW

File: packages/modules/init-capital/src/actions/deployApi3ProxyOracleReader.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  constructor(data: deployApi3ProxyOracleReaderActionData) {
+    validateActionData(data, DeployApi3ProxyOracleReaderActionParamsSchema, ['deployer'])
+    super(DeployApi3ProxyOracleReaderAction.name, data)
+  }
+  protected getSubActions(): SubAction[] {
+    const deployer = this.data.signer['deployer']
+    const params = this.data.params
+
+    // return subactions
+    return [new DeployApi3ProxyOracleReaderSubAction(deployer, params)]
```
Comment (Kittikornl) [inline packages/modules/init-capital/src/actions/deployApi3ProxyOracleReader.ts:34]: we deploy this as proxy pls add more subaction

File: packages/modules/init-capital/src/actions/subactions/deployApi3ProxyOracleReaderProxy.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  public async updateRegistryAndMessage(
+    registry: InitCapitalRegistry,
+    txHashes: Hex[],
+  ): Promise<SubActionExecuteResponse<InitCapitalRegistry>> {
+    if (txHashes.some((v) => !v)) {
+      throw new TxNotFoundError()
+    }
+
+    const [deployInitCoreProxyHash] = txHashes
```
Comment (Kittikornl) [inline packages/modules/init-capital/src/actions/subactions/deployApi3ProxyOracleReaderProxy.ts:39]: rename

File: packages/modules/init-capital/src/actions/subactions/deployApi3ProxyOracleReader.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+export type DeployApi3ProxyOracleReaderSubActionParams = DeployApi3ProxyOracleReaderTxBuilderParams
+
+export type DeployApi3ProxyOracleReaderMsg = {
+  api3ProxyOracleReaderImpl: Address
+}
+
+export class DeployApi3ProxyOracleReaderSubAction extends SubAction<
+  DeployApi3ProxyOracleReaderSubActionParams,
+  InitCapitalRegistry,
+  object
```
Comment (tapakornl) [inline packages/modules/init-capital/src/actions/subactions/deployApi3ProxyOracleReader.ts:22]: is this the correct msg type?

File: packages/modules/init-capital/src/actions/subactions/deployApi3ProxyOracleReaderProxy.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import { DeployTransparentUpgradeableProxyTxBuilder } from '@actions/subactions/tx-builders/TransparentUpgradeableProxy/deploy'
+
+import { InitCapitalRegistry } from '@/src/type'
+
+export type DeployApi3ProxyOracleProxySubActionParams = {
+  proxyAdmin: Address
+  api3ProxyOracleReaderImpl: Address
+}
+
+export class DeployApi3ProxyOracleProxySubAction extends SubAction<DeployApi3ProxyOracleProxySubActionParams, InitCapitalRegistry> {
```
Comment (tapakornl) [inline packages/modules/init-capital/src/actions/subactions/deployApi3ProxyOracleReaderProxy.ts:15]: add msg type

**Review by Kittikornl — APPROVED** [review]

**Review by Kittikornl — APPROVED** [review]

**Pull request merged**